src/Raven.Client/Document/Async/AsyncDocumentSession.cs

```diff
@@ -712,44 +710,32 @@ public Task<T[]> LoadAsync<T>(IEnumerable<ValueType> ids, CancellationToken toke
         /// <returns></returns>
         public async Task<T> LoadAsync<T>(string id, CancellationToken token = default(CancellationToken))
         {
-            if (id == null)
-                throw new ArgumentNullException("id", "The document id cannot be null");
-            object entity;
-            if (entitiesByKey.TryGetValue(id, out entity))
-            {
-                return (T)entity;
-            }
-            JsonDocument value;
-            if (includedDocumentsByKey.TryGetValue(id, out value))
+            var loadOeration = new NewLoadOperation(this);
+            loadOeration.ById(id);
+
+            var command = loadOeration.CreateRequest();
+            if (command != null)
             {
-                includedDocumentsByKey.Remove(id);
-                return TrackEntity<T>(value);
+                await RequestExecuter.ExecuteAsync(command, Context);
+                loadOeration.SetResult(command.Result);
             }
-            if (IsDeleted(id))
-                return default(T);
 
-            IncrementRequestCount();
-            var loadOperation = new LoadOperation(this, AsyncDatabaseCommands.DisableAllCaching, id);
-            return await CompleteLoadAsync<T>(id, loadOperation, token).ConfigureAwait(false);
+            return loadOeration.GetDocument<T>();
         }
 
-        private async Task<T> CompleteLoadAsync<T>(string id, LoadOperation loadOperation, CancellationToken token = default(CancellationToken))
+        public async Task<T[]> LoadAsync<T>(IEnumerable<string> ids, CancellationToken token = default(CancellationToken))
         {
-            loadOperation.LogOperation();
-            using (loadOperation.EnterLoadContext())
-            {
-                var result = await AsyncDatabaseCommands.GetAsync(id, token: token).ConfigureAwait(false);
-
-                if (loadOperation.SetResult(result) == false)
-                    return loadOperation.Complete<T>().FirstOrDefault();
+            var loadOeration = new NewLoadOperation(this);
+            loadOeration.ByIds(ids);
 
-                return await CompleteLoadAsync<T>(id, loadOperation, token).WithCancellation(token).ConfigureAwait(false);
+            var command = loadOeration.CreateRequest();
+            if (command != null)
+            {
+                await RequestExecuter.ExecuteAsync(command, Context);
+                loadOeration.SetResult(command.Result);
             }
-        }
 
-        public Task<T[]> LoadAsync<T>(IEnumerable<string> ids, CancellationToken token = default(CancellationToken))
-        {
-            return LoadAsyncInternal<T>(ids.ToArray(), token);
+            return loadOeration.GetDocuments<T>();
```

Comment:
what will be the order here? it must be the same as specified in the LoadAsync ids parameter and IIRC it must support duplicates